lib/david/transmitter.rb

```diff
@@ -6,7 +6,7 @@ class Transmitter
 
     def initialize(socket)
       @log = Celluloid.logger
-      @socket = socket || server.socket
+      @socket = socket
```

Comment:
So far I didn't manage to find a function which calls this constructor with nil as an argument. @nning are you aware of any functions which does this?